addons/hr_expense/static/src/mixins/document_upload.js

```diff
@@ -88,11 +91,26 @@ export const ExpenseDocumentUpload = (T) => class ExpenseDocumentUpload extends
     }
 
     uploadDocument() {
+        this.uploadsProcessing++;
         this.fileInput.el.click();
     }
 
     async onChangeFileInput() {
-        await this._onChangeFileInput([...this.fileInput.el.files]);
+        try {
+            await this._onChangeFileInput([...this.fileInput.el.files]);
+            if (this.uploadsProcessing === 1) {
+                await this.actionService.doAction({
```

Comment:
I'm not sure how it could be solved, but a small UX issue I see is that when I upload a document, wait for it to be processed, it leads to a list view "Generate Expenses". Let's imagine I forgot another document, and I therefore press Upload again. When finished processing, it opens a new list view (a child in the breadcrumb) with this new expense. Wouldn't it be better to simply add it to the list if we are already in the list view ('Generate Expenses') ? :thinking: